### ISSUE TITLE
[BUG] a lot of errors during start `Call to function 'CDevice_update' failed, exception details:`

### ISSUE THREAD

**Issue by Rouzax** [ajarzyn/domoticz-luxtronic2#7]
With the latest version I get a lot of errors during the start of the plugin, it does seem to get all the info though
```
 2022-07-31 10:16:36.860 Error: WP: (CDevice_update) Heating supply temperature: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.860 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.875 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.875 Error: WP: (CDevice_update) Heating return temperature: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.875 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.876 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.876 Error: WP: (CDevice_update) Heating return temperature - target: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.876 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.876 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.876 Error: WP: (CDevice_update) Outside temperature: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.876 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.876 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.876 Error: WP: (CDevice_update) Outside temperature - average: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.876 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.876 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.876 Error: WP: (CDevice_update) Hot water temperature: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.876 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.876 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.876 Error: WP: (CDevice_update) Ground source temperature - in: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.876 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.876 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.876 Error: WP: (CDevice_update) Ground source temperature - out: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.876 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.876 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.876 Error: WP: (CDevice_update) OM 1 Temperature: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.876 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.876 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.876 Error: WP: (CDevice_update) OM 1 Temperature - target: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.876 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.876 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.876 Error: WP: (CDevice_update) Energy produced - heating: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.876 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.876 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.876 Error: WP: (CDevice_update) Energy produced - hot water: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.876 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.876 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.876 Error: WP: (CDevice_update) Energy produced - sum: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.876 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.876 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.876 Error: WP: (CDevice_update) Heating mode: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.876 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.876 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.877 Error: WP: (CDevice_update) Hot water mode: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.877 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.877 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.877 Error: WP: (CDevice_update) Cooling: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.877 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.877 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.877 Error: WP: (CDevice_update) Temperature +-: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.877 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.877 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.881 Error: WP: (CDevice_update) Hot water temperature - target: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.881 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.881 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.888 Error: WP: (CDevice_update) Working mode: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.888 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.888 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.888 Error: WP: (CDevice_update) Flow: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.888 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.888 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.888 Error: WP: (CDevice_update) Compressor frequency: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.888 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.888 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.888 Error: WP: (CDevice_update) Actual room temperature: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.888 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.888 Error: WP: Exception: 'TypeError'. No traceback available.
2022-07-31 10:16:36.888 Error: WP: (CDevice_update) Room temperature set: Failed to parse parameters: 'nValue', 'sValue', 'Image', 'SignalLevel', 'BatteryLevel', 'Options', 'TimedOut', 'Name', 'TypeName', 'Type', 'Subtype', 'Switchtype', 'Used', 'Description', 'Color' or 'SuppressTriggers' expected.
2022-07-31 10:16:36.889 Error: WP: Call to function 'CDevice_update' failed, exception details:
2022-07-31 10:16:36.889 Error: WP: Exception: 'TypeError'. No traceback available. 
```

**Comment by Rouzax** [comment]
During a fetch it does get all info
```
2022-07-31 10:23:20.016 WP: Pushing 'onHeartbeatCallback' on to queue
2022-07-31 10:23:20.066 WP: Processing 'onHeartbeatCallback' message
2022-07-31 10:23:20.066 WP: Acquiring GIL for 'onHeartbeatCallback'
2022-07-31 10:23:20.066 WP: Calling message handler 'onHeartbeat' on 'module' type object.
2022-07-31 10:23:20.066 WP: onHeartbeat called.
2022-07-31 10:23:20.072 WP: update_device unit: 1 Name: Heating supply temperature with parameters: {'nValue': 0, 'sValue': '25.3'}
2022-07-31 10:23:20.072 (Heating supply temperature) Updating device from 0:'25.3' to have values 0:'25.3'.
2022-07-31 10:23:20.082 WP: update_device unit: 2 Name: Heating return temperature with parameters: {'nValue': 0, 'sValue': '23.4'}
2022-07-31 10:23:20.082 (Heating return temperature) Updating device from 0:'23.4' to have values 0:'23.4'.
2022-07-31 10:23:20.093 WP: update_device unit: 3 Name: Heating return temperature - target with parameters: {'nValue': 0, 'sValue': '19.8'}
2022-07-31 10:23:20.093 (Heating return temperature - target) Updating device from 0:'19.8' to have values 0:'19.8'.
2022-07-31 10:23:20.105 WP: update_device unit: 4 Name: Outside temperature with parameters: {'nValue': 0, 'sValue': '21.5'}
2022-07-31 10:23:20.105 (Outside temperature) Updating device from 0:'21.5' to have values 0:'21.5'.
2022-07-31 10:23:20.127 WP: update_device unit: 5 Name: Outside temperature - average with parameters: {'nValue': 0, 'sValue': '23.0'}
2022-07-31 10:23:20.127 (Outside temperature - average) Updating device from 0:'23.0' to have values 0:'23.0'.
2022-07-31 10:23:20.148 WP: update_device unit: 6 Name: Hot water temperature with parameters: {'nValue': 0, 'sValue': '57.9'}
2022-07-31 10:23:20.148 (Hot water temperature) Updating device from 0:'57.9' to have values 0:'57.9'.
2022-07-31 10:23:20.160 WP: update_device unit: 7 Name: Ground source temperature - in with parameters: {'nValue': 0, 'sValue': '20.4'}
2022-07-31 10:23:20.160 (Ground source temperature - in) Updating device from 0:'20.4' to have values 0:'20.4'.
2022-07-31 10:23:20.171 WP: update_device unit: 8 Name: Ground source temperature - out with parameters: {'nValue': 0, 'sValue': '20.5'}
2022-07-31 10:23:20.171 (Ground source temperature - out) Updating device from 0:'20.3' to have values 0:'20.5'.
2022-07-31 10:23:20.183 WP: update_device unit: 9 Name: OM 1 Temperature with parameters: {'nValue': 0, 'sValue': '75.0'}
2022-07-31 10:23:20.184 (OM 1 Temperature) Updating device from 0:'75.0' to have values 0:'75.0'.
2022-07-31 10:23:20.193 WP: update_device unit: 10 Name: OM 1 Temperature - target with parameters: {'nValue': 0, 'sValue': '0.0'}
2022-07-31 10:23:20.193 (OM 1 Temperature - target) Updating device from 0:'0.0' to have values 0:'0.0'.
2022-07-31 10:23:20.202 WP: update_device unit: 11 Name: Energy produced - heating with parameters: {'nValue': 0, 'sValue': '0;0.0'}
2022-07-31 10:23:20.202 (Energy produced - heating) Updating device from 0:'0;0.0' to have values 0:'0;0.0'.
2022-07-31 10:23:20.213 WP: update_device unit: 12 Name: Energy produced - hot water with parameters: {'nValue': 0, 'sValue': '0;169900.0'}
2022-07-31 10:23:20.213 (Energy produced - hot water) Updating device from 0:'0;169900.0' to have values 0:'0;169900.0'.
2022-07-31 10:23:20.223 WP: update_device unit: 13 Name: Energy produced - sum with parameters: {'nValue': 0, 'sValue': '0;174600.0'}
2022-07-31 10:23:20.223 (Energy produced - sum) Updating device from 0:'0;174600.0' to have values 0:'0;174600.0'.
2022-07-31 10:23:20.245 WP: 5 [(3, 'Heating mode'), (4, 'Hot water mode'), (2, 'Swimming pool mode / Photovaltaik'), (2, 'EVUM'), (1, 'Defrost'), (0, 'No requirement'), (4, 'Heating external source mode'), (1, 'Cooling')]
2022-07-31 10:23:20.245 WP: update_device unit: 19 Name: Working mode with parameters: {'nValue': 0, 'sValue': 'No requirement'}
2022-07-31 10:23:20.245 (Working mode) Updating device from 0:'No requirement' to have values 0:'No requirement'.
2022-07-31 10:23:20.267 WP: update_device unit: 20 Name: Flow with parameters: {'nValue': 0, 'sValue': '0.0'}
2022-07-31 10:23:20.267 (Flow) Updating device from 0:'0.0' to have values 0:'0.0'.
2022-07-31 10:23:20.278 WP: update_device unit: 21 Name: Compressor frequency with parameters: {'nValue': 0, 'sValue': '0.0'}
2022-07-31 10:23:20.278 (Compressor frequency) Updating device from 0:'0.0' to have values 0:'0.0'.
2022-07-31 10:23:20.288 WP: update_device unit: 22 Name: Actual room temperature with parameters: {'nValue': 0, 'sValue': '21.1'}
2022-07-31 10:23:20.288 (Actual room temperature) Updating device from 0:'21.1' to have values 0:'21.1'.
2022-07-31 10:23:20.297 WP: update_device unit: 23 Name: Room temperature set with parameters: {'nValue': 0, 'sValue': '21.5'}
2022-07-31 10:23:20.297 (Room temperature set) Updating device from 0:'21.5' to have values 0:'21.5'.
2022-07-31 10:23:20.324 WP: update_device unit: 14 Name: Heating mode with parameters: {'nValue': 0, 'sValue': '0'}
2022-07-31 10:23:20.324 (Heating mode) Updating device from 0:'0' to have values 0:'0'.
2022-07-31 10:23:20.337 WP: update_device unit: 15 Name: Hot water mode with parameters: {'nValue': 0, 'sValue': '0'}
2022-07-31 10:23:20.337 (Hot water mode) Updating device from 0:'0' to have values 0:'0'.
2022-07-31 10:23:20.350 WP: update_device unit: 16 Name: Cooling with parameters: {'nValue': 1, 'sValue': ''}
2022-07-31 10:23:20.350 (Cooling) Updating device from 1:'' to have values 1:''.
2022-07-31 10:23:20.376 WP: update_device unit: 17 Name: Temperature +- with parameters: {'nValue': 110, 'sValue': '110'}
2022-07-31 10:23:20.376 (Temperature +-) Updating device from 110:'110' to have values 110:'110'.
2022-07-31 10:23:20.390 WP: update_device unit: 18 Name: Hot water temperature - target with parameters: {'nValue': 600, 'sValue': '600'}
2022-07-31 10:23:20.390 (Hot water temperature - target) Updating device from 600:'600' to have values 600:'600'.
2022-07-31 10:23:20.419 WP: Acquiring GIL for 'onHeartbeatCallback'
```

**Comment by ajarzyn** [comment]
> With the latest version

You mean Domoticz or plugin?

**Comment by Rouzax** [comment]
I meant the plugin although Domoticz is also on the latest dev branch

**Comment by ajarzyn** [comment]
You forced me to finally upgrade my Domoticz. (Last update 14months ago) thanks for that.

Which version of Domoticz do you have?

I did check with last stable version from January: https://domoticz.com/downloads/
  | 2022.1.14127 |   | 2022-01-31
And I've observed similar problem but apart from problem you mentioned there were many more and I just wasn't interested in checking that image.

I've upgraded to beta release:
  | 2022.1.14440 |   | 2022-07-30
: 2022.1 (build 14440)
: b793752ee
: 2022-07-30 17:12:25

And I did not observed the problem. Try upgrading domoticz.

**Comment by ajarzyn** [comment]
Ok I was able to reproduce it.
I will investigate it but it feels like problem with Domoticz.

**Comment by ajarzyn** [comment]
Duplicate of #4

**Comment by Rouzax** [comment]
> You forced me to finally upgrade my Domoticz. (Last update 14months ago) thanks for that.
> 
> Which version of Domoticz do you have?
> 
> I did check with last stable version from January: https://domoticz.com/downloads/   | 2022.1.14127 |   | 2022-01-31 And I've observed similar problem but apart from problem you mentioned there were many more and I just wasn't interested in checking that image.
> 
> I've upgraded to beta release:   | 2022.1.14440 |   | 2022-07-30 : 2022.1 (build 14440) : b793752ee : 2022-07-30 17:12:25
> 
> And I did not observed the problem. Try upgrading domoticz.

I'm on the latest dev branch 14440 build.
Reason is that I'm using Zwave and want to switch from OpenZwave to ZwaveJS2MQTT

**Comment by ajarzyn** [comment]
Fix delivered with [fe1d66c](https://github.com/ajarzyn/domoticz-luxtronic2/commit/fe1d66c96c301f25433d30fb44741068515a0c24)
